gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/PubSub.java

```diff
@@ -94,18 +99,54 @@ public static ListOption pageToken(String pageToken) {
       Integer getInteger(Map<Option.OptionType, ?> options) {
         return get(options);
       }
+
+      ExecutorFactory<ExecutorService> getExecutorFactory(Map<Option.OptionType, ?> options) {
+        return get(options);
+      }
     }
 
     private PullOption(Option.OptionType option, Object value) {
       super(option, value);
     }
 
     /**
-     * Returns an option to specify the maximum number of messages that can be processed
-     * concurrently at any time.
+     * Returns an option to specify the maximum number of messages that can be queued in the message
+     * consumer at any time. Queued messages are already pulled messages that are either waiting to
+     * be processed or being processed. Queued messages will have their acknowledge deadline renewed
+     * until they are acknowledged or "nacked". If not provided, at most 100 messages can be in the
+     * queue.
      */
-    public static PullOption maxConcurrentCallbacks(int maxConcurrency) {
-      return new PullOption(OptionType.MAX_CONCURRENT_CALLBACKS, maxConcurrency);
+    public static PullOption maxQueuedCallbacks(int maxQueuedCallbacks) {
+      return new PullOption(OptionType.MAX_QUEUED_CALLBACKS, maxQueuedCallbacks);
+    }
+
+    /**
+     * Returns an option to specify the executor used to execute message processor callbacks. The
+     * executor determines the number of messages that can be processed at the same time. The
+     * {@code shouldAutoClose} parameter sets whether the executor should be shutdown when the
+     * message consumer is closed. If not provided, a single-threaded executor is used.
+     *
+     * @param executor the executor used to run message processor callbacks
+     * @param shouldAutoClose if {@code true}, the executor is shutdown when the message consumer is
```

Comment:
Nit, I am not sure if 'Close' is better than 'Shutdown' for this variable? 'autoShutdown'/'shutdown'/...